### ISSUE TITLE
Possible bugs/corrections to abasic_v10c s19 file based on disassembly, cassette version from flex disk,  and my orig notes

### ISSUE THREAD

**Issue by stanleyruppert** [linuxha/RT68mx#2]
Hi Neil,
Now that VCF is over, I'm taking a fresh look at rtedit and microware abasic.
I'm dedicating one of my D2 kits for the implementation as well as a M68MM01A2 micromodule with AC-30 cassette attached.

Kudos for the start at disassembly! 
In looking at my original 1970s margin notes, a cassette version (binary from CMD file) from flexdisk archives, and your disassembly, we might consider the following fixes.

My original manual fades on the abasic listings from $0F10 to $0F30. I can't read the s19 record checksums anymore. 
From my vintage notes I had written down the databytes as I originally gave you.

Currently in your git: (with checksums I computed before sending you)
```
S1130F1002973EDE315FD72CD729D72DD751D75230. - readable in manual with checksum $30. Note 97 $3E
S1130F2039DE318DD48D0B240896522603BD10FA78 - very faint in manual, computed checksum from databytes in my notes
S1130F300D398D378D2E812B26088D2B5FBD10FC2E- very faint in manual, computed checksum from databytes in my notes
```

Looking at abasic.bin (converted from ABASIC.CMD within the FLEX2-01 disk image) - V1.0c compiled for 12K and slight modifications for flex OS

We see the following:
```
S1130F1002974EDE315FD72CD729D72DD751D75220     <---note $4E instead of $3E. $3E is utilized by 16 bit  index reg instructions elsewhere in code
Make more sense to store accumulator in $4E as with other code, but unclear given weird coding practices microware used.
checksum is computed to be consistent with changed value to $4E

S1130F2039DE318DD48D0B240896522603BD10FA78    <--- identical in both manual version and Flex image

S1130F300D398D378D2E812B26088D2B5FBD10BC6E   <---note BD $10BC and not BD $10FC. Disassembly suggests $10BC correct. Could be error in my original vintage notes. I no longer have readable cassette tapes of the object code to confirm.
checksum is computed.
```

so bottom line - we should probably use $10BC as JSR target. Further investigation will be needed for the $97 (STAA direct) $3E vs $4E.

Stan

PS, when I clicked on your youtube link for my vcf talk, the timestamp started just after my interview. Might adjust.
I got raw video files from another attendee who interviewed me w/B-roll footage. I'm editing and will send you a link when done.

screenshot of binary differences either side of $0F10  ($0E10 in screenshot as file offset 0x0100)
![Screenshot 2024-08-20 at 1 14 45 PM](https://github.com/user-attachments/assets/4f106f45-d6e7-4366-a6e3-f93792cd9bfa)

**Comment by linuxha** [comment]
On 8/20/24 16:20, stanleyruppert wrote:
> Hi Neil,
> Now that VCF is over, I'm taking a fresh look at rtedit and microware abasic.
> I'm dedicating one of my D2 kits for the implementation as well as a M68MM01A2 micromodule with AC-30 cassette attached.

Saw the raw video! I want to see and hear more :-)

Going for a bike ride, be back later but I've been busy. Still not getting things
to work found a whole lot of interesting (ABASIC Flex can built RT68MX code) and
ABASIC 3.0 (???).

I purchased  PT68-1 and that where I have ABASICFL 2.1. The ABASIC 3.0 is in
the ABASIC-FLEX dir.

I'll try to explain later the rest.

--
Linux Home Automation         Neil Cherry       ***@***.***
http://www.linuxha.com/                         Main site
http://linuxha.blogspot.com/                    My HA Blog
Author of:    	Linux Smart Homes For Dummies   KD2ZRQ /AG

**Comment by linuxha** [comment]
I'll need to reply tomorrow, need to get to bed early (grrr).

**Comment by stanleyruppert** [comment]
Hey no worries. No rush. Give me a shout when convenient.

Stan

**Comment by linuxha** [comment]
I see 0F10 very clearly in the manual as:

```
S1130F1002973EDE315FD72CD729D72DD751D75230
```

BUT! I don't have 0F30 in my copy of the RT68-ABASIC-RTEDIT.pdf (???). Do we have 2 different copies of the manual?

But I've create a new file with your recommendations, the abasic-10c-16K-B.s19 (I have more than 16K free)
abasic-10c-16K-C.s19 has the 0F30 changes (make perfect sense) but not the 0F10 changes (you're arguement makes sense but it's clear in my manual).

**Comment by linuxha** [comment]
Oh found Microware ads from 1979. Seems Microware had A/D, D/A interfaces, a 6800 Chess, a LISP Interpreter, Dr Eliza, A/BASIC Compiler (Disk and cassette version), A/BASIC Source Generator (I had not heard of this), A/BASIC Interpreter, RT68MX (6830 ROM) and RT68MXP (2708 ROM). The Disk version was available for Flex/Mini-Flex, and Smoke Signals Broadcast DOS-68. The ads mention mini-disk (5.25") but not Flex directly.

**Comment by stanleyruppert** [comment]
Only one copy of the manual - my original manual that I scanned and uploaded to deramp and Mike Evenson's site. You'll note that lines for 0B90 (faded) and 0BA0 is missing as well as 0F20/30. Also $19E0 line is unreadable. These all came from my 45 year old margin notes. I got those bytes from original tape I had back in the 70s. I misread my scribbled ancient comments when typing in the 0F30 bytes into the s19 text file. 0F10 is clear, but it could still be a bug I didn't notice  (or failed to write down if I patched it way back then). We'll have to try both and see what is ultimately correct. Odd that it is 4E on the cassette 1.0C versions within the flex disk image but 3E in the manual. Listings have been known to contain errors...
My original microware tapes, as well as my working tapes with software and tweaks, unfortunately were no longer recoverable. Spent too long in various basements / attics. All my original hardware still works - but paper is the ultimate long lasting archive media...I have RT68 and RTedit running now on my M68MM01A2 board. I've patched per that kilobaud article to skip the tape IO and leave text source in memory. Next up is trying Abasic with the patches to get source from memory as well.

**Comment by linuxha** [comment]
Excellent, this is progress. :-)

abasic-10c-16K-B.s19 should be useful. I'll try to get the 12K version updated tomorrow.
